[receiver/sshcheck] Include SSH endpoint resource attribute

DewaldDeJager opened the pull request:
**Description:** The metadata.yml for the SSH check receiver currently documents a resource attribute containing the SSH endpoint but this is not emitted. This PR updates the receiver to include this resource attribute.

**Link to tracking Issue:** #24441 

**Testing:**

Example collector config:
```yaml
receivers:
  sshcheck:
    endpoint: 13.245.150.131:22
    username: ec2-user
    key_file: /Users/dewald.dejager/.ssh/sandbox.pem
    collection_interval: 15s
    known_hosts: /Users/dewald.dejager/.ssh/known_hosts
    ignore_host_key: false
    resource_attributes:
      "ssh.endpoint":
        enabled: true

exporters:
  logging:
    verbosity: detailed
  prometheus:
    endpoint: 0.0.0.0:8081
    resource_to_telemetry_conversion:
      enabled: true

service:
  pipelines:
    metrics:
      receivers: [sshcheck]
      exporters: [logging, prometheus]
```

The log output looks like this:
```
2023-07-30T16:52:38.724+0200    info    MetricsExporter {"kind": "exporter", "data_type": "metrics", "name": "logging", "resource metrics": 1, "metrics": 2, "data points": 2}
2023-07-30T16:52:38.724+0200    info    ResourceMetrics #0
Resource SchemaURL: 
Resource attributes:
     -> ssh.endpoint: Str(13.245.150.131:22)
ScopeMetrics #0
ScopeMetrics SchemaURL: 
InstrumentationScope otelcol/sshcheckreceiver 0.82.0-dev
Metric #0
Descriptor:
     -> Name: sshcheck.duration
     -> Description: Measures the duration of SSH connection.
     -> Unit: ms
     -> DataType: Gauge
NumberDataPoints #0
StartTimestamp: 2023-07-30 14:52:22.381672 +0000 UTC
Timestamp: 2023-07-30 14:52:38.404003 +0000 UTC
Value: 319
Metric #1
Descriptor:
     -> Name: sshcheck.status
     -> Description: 1 if the SSH client successfully connected, otherwise 0.
     -> Unit: 1
     -> DataType: Sum
     -> IsMonotonic: false
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
StartTimestamp: 2023-07-30 14:52:22.381672 +0000 UTC
Timestamp: 2023-07-30 14:52:38.404003 +0000 UTC
Value: 1
```

And the Prometheus metrics look like this:
```
# HELP sshcheck_duration Measures the duration of SSH connection.
# TYPE sshcheck_duration gauge
sshcheck_duration{ssh_endpoint="13.245.150.131:22"} 311
# HELP sshcheck_status 1 if the SSH client successfully connected, otherwise 0.
# TYPE sshcheck_status gauge
sshcheck_status{ssh_endpoint="13.245.150.131:22"} 1
```

DewaldDeJager committed:
Emit SSH endpoint resource attribute (#24441)

DewaldDeJager committed:
Emit SSH endpoint resource attribute (#24441)

DewaldDeJager committed:
Merge remote-tracking branch 'origin/feature/emitSshEndpointResourceAttribute' into feature/emitSshEndpointResourceAttribute

DewaldDeJager committed:
Remove change to CHANGELOG

DewaldDeJager commented:
The failing test is unrelated to the changes in this PR. Looks like a K8s scheduling/resourcing issue. Please would you trigger a re-run of the failing check?

atoulme commented:
Done.

atoulme commented on receiver/sshcheckreceiver/testdata/config.yaml:
```diff
@@ -3,7 +3,7 @@ receivers:
     username: notdefault_password
     password: notdefault_password
-    keyfile: notdefault/path/keyfile
+    key_file: notdefault/path/keyfile
```
Is this fix related?

DewaldDeJager replied:
I needed to use this test data to test the change and noted that it was outdated. Should I update it in a different PR?

atoulme replied:
Maybe best. Interesting that you can change this and no tests failed. Making it a separate pr or issue would force us to look at validating that file going forward.

DewaldDeJager replied:
Created #24739 for this

atoulme commented on receiver/sshcheckreceiver/scraper.go:
```diff
@@ -124,13 +125,15 @@ func (s *sshcheckScraper) scrape(ctx context.Context) (_ pmetric.Metrics, err er
 	}
 
-	return s.mb.Emit(), nil
+	s.rb.SetSSHEndpoint(s.Config.SSHClientSettings.Endpoint)
```
Do you mind adding a test checking this gets indeed propagated?

DewaldDeJager replied:
Done

DewaldDeJager committed:
Add test to check that resource attribute is propagated to metrics

DewaldDeJager committed:
Add test to check that resource attribute is propagated to metrics


Add test to check that resource attribute is propagated to metrics


Revert the change to the test config file

DewaldDeJager committed:
Merge remote-tracking branch 'origin/feature/emitSshEndpointResourceAttribute' into feature/emitSshEndpointResourceAttribute

DewaldDeJager commented:
One of the lint matrix checks is failing because of a timeout. I don't think it's related to these changes. Please could you re-run that check?
